tests/pylint/test_enforce_type_hints.py

```diff
@@ -469,3 +469,43 @@ def async_get_options_flow(
 
     with assert_no_messages(linter):
         type_hint_checker.visit_classdef(class_node)
+
+
+def test_invalid_lock_entity(
+    linter: UnittestLinter, type_hint_checker: BaseChecker
+) -> None:
+    """Ensure invalid hints are rejected for LockEntity properties."""
```

Comment:
Does it make sense to rename that test case? It might be better to just have one making sure properties are only checked with `ignore_missing_annotations = False`. Essentially what this test case already does.